br/pkg/lightning/backend/external/engine.go

```diff
@@ -293,9 +293,9 @@ func (e *Engine) LoadIngestData(
 	regionRanges []common.Range,
 	outCh chan<- common.DataAndRange,
 ) error {
-	// currently we assume the region size is 96MB and will download 96MB*40 = 3.8GB
+	// currently we assume the region size is 96MB and will download 96MB*20 = 1.9GB
 	// data at once
-	regionBatchSize := 40
+	regionBatchSize := 20
```

Comment:
I have tested that only with this change, performance won't drop and OOM can be avoided with "thread=8" SQL parameter.

Reply:
In this week when I run the 1T test with thread=8 and thread=16 arguments, thread=8 will OOM more frequently.

Considering that before this PR, merge step will loop for every 1000 files merging them into 8 or 16 files. When thread=8, one output file is merged from 125 files, when thread=16, one output file is merged from 63 files. OOM may cause by the larger ourput file size. And lower this batch size will reduce the chance of OOM.

Now we are going to let the merge step loop for 4000 files, with thread=16, one output file is merged from 250 files. So it's more likely that OOM will occur. I'm running the 1T test to check if the batch size is lower enough.